snippets/scripts/compass-github-importer/index.js

```diff
@@ -0,0 +1,174 @@
+const axios = require('axios');
+const chalk = require('chalk')
+require('dotenv').config()
+
+const {GITHUB_URL, ACCESS_TOKEN, USER_EMAIL, TOKEN, TENANT_SUBDOMAIN, CLOUD_ID} = process.env
+const dryRun = process.env.DRY_RUN === "1"
+
+const ATLASSIAN_GRAPHQL_URL = `https://${TENANT_SUBDOMAIN}.atlassian.net/gateway/api/graphql`;
+
+function makeGqlRequest(query) {
+    const header = btoa(`${USER_EMAIL}:${TOKEN}`);
+    return fetch(ATLASSIAN_GRAPHQL_URL, {
+        method: 'POST',
+        headers: {
+            Authorization: `Basic ${header}`,
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(query),
+    }).then((res) => res.json());
+}
+
+// This function looks for a component with a certain name, but if it can't find it, it will create a component.
+async function putComponent(componentName, description, web_url) {
+    const response = await makeGqlRequest({
+        query: `
+      query getComponent {
+        compass @optIn(to: "compass-beta") {
+          searchComponents(cloudId: "${CLOUD_ID}", query: {
+            query: "${componentName}",
+            first: 1
+          }) {
+            ... on CompassSearchComponentConnection {
+              nodes {
+                component {
+                  id
+                  name
+                }
+              }
+            }
+          }
+        }
+      }
+      `,
+    });
+    const maybeResults = response?.data?.compass?.searchComponents?.nodes;
+
+    if (!Array.isArray(maybeResults)) {
+        console.error(`error fetching component: `, JSON.stringify(response));
+        throw new Error('Error fetching component');
```

Comment:
Do we want to throw with some additional details here like the componentId, organizationId (non UGC data that is available)?

Reply:
Good, call will do!